cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -241,6 +241,32 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
 				Region: options.Region,
 			}))
 
+			osbuildJobResult.Success = true
+			osbuildJobResult.UploadStatus = "success"
+		case *target.AWSS3TargetOptions:
```

Comment:
Might even want to introduce `jobimpl-osbuild-s3.go` and make this a whole new job type, specific to S3.

Reply:
Not sure I follow exactly.  How much of a job type of its own will this be?  I can see how splitting it out of `target/aws_target.go` into its own `target/s3_target.go` would make sense, but I don't see separate jobs for the other targets.

Reply:
Ah, you need to inspect the contents of my brain that I didn't include here, sorry. We had an idea way back when to gradually deprecate the `osbuild` job type, and replace it with things in the style of `osbild-koji` and friends. Idea being that building/pushing is one job type and "registering" is another. And rather than having the target enum the arguments are just part of the job type.

That should probably be a separate discussion from this PR though, this is good as is, just wanted to mention it as something to keep in mind/discuss eventually.

Reply:
ah, that's a nice plan.  I don't know if it makes sense to even plant the seed of that here though.  I'm worried we might get sidetracked a bit too far.

Reply:
Yeah, fair enough. Let's ignore that for now!